server/test/suites/categoryRouter.test.js

```diff
@@ -166,4 +166,100 @@ module.exports = function categorySuite() {
         });
     });
   });
+
+  describe('/getCategoryItems/:categoryId', () => {
+    before(async () => {
+      await item1.save();
+      await item2.save();
+    });
+
+    after(async () => {
+      await Item.findByIdAndDelete(item2._id);
+    });
+
+    it('/getCategoryItems SetUp test: 3 categories in Category db, 2 items in Item db', async () => {
+      const categoryCount = await Category.countDocuments();
+      expect(categoryCount).equals(3);
+      const itemCount = await Item.countDocuments();
+      expect(itemCount).equals(2);
+    });
+
+    it('get items in testCategory1, should return response.body.items with 2 items', (done) => {
+      chai
+        .request(server)
+        .get('/category/getCategoryItems/' + testCategory1._id)
+        .end((error, response) => {
+          expect(response).to.have.status(200);
+          expect(response.body.title).equals(testCategory1.title);
+          expect(response.body.items).to.have.lengthOf(2);
+          expect(response.body.items[0]._id.toString()).equals(item1._id.toString());
+          expect(response.body.items[1]._id.toString()).equals(item2._id.toString());
```

Comment:
not sure if this is necessary but do we wanna also check if the content of the items are as expected (source, highlighted text, etc..). If we do that, it'd be good to have a function looping through all those things and you can just call that function.

Reply:
discussed offline